src/cmds/pbsdsh.c

```diff
@@ -95,6 +107,96 @@ bailout(int sig)
  * @return - Void
  *
  */
+
+
+int
+host_match(char *line, char *host)
+{
+	if (NULL == host ) return 0;
+	int len = strlen(line);
+	static char domain[PBS_MAXHOSTNAME + 1];
+	char fullhost[PBS_MAXHOSTNAME + 1];
+	static struct in_addr addr;
+	static int addrvalid = -1;
+
+	if (line[len - 1] == '\n')
+		line[len - 1] = '\0';
+
+	if (strcmp(line, host) == 0)
+		return 1;
+
+	if (addrvalid == -1) {
+		addr.s_addr = inet_addr(host);
+		addrvalid = (addr.s_addr == INADDR_NONE) ? 0 : 1;
+	}
+
+	if (addrvalid) { /* compare IP addresses */
+		struct hostent *hp = gethostbyname(line);
+		int i;
+
+		if (hp == NULL)
+			return 0;
+
+		for (i = 0; hp->h_addr_list[i]; i++) {
+			if (memcmp(&addr, hp->h_addr_list[i],
+				   hp->h_length) == 0)
+				return 1;
+		}
+		return 0;
+	}
+
+	if (domain[0] == '\0') {
+		if (getdomainname(domain, (sizeof(domain) - 1)) == -1) {
+			perror("getdomainname");
+			exit(255);
+		}
+		if (domain[0] == '\0') {
+			int i;
+			char *dot;
+
+			if (gethostname(domain, (sizeof(domain) - 1)) == -1) {
+				perror("getdomainname");
+				exit(255);
+			}
+			if (domain[0] == '\0')
+				return 0;
+			if ((dot = strchr(domain, '.')) == NULL)
+				return 0;
+			for (i = 0, dot++; *dot; i++, dot++)
+				domain[i] = *dot;
+			domain[i] = '\0';
+		}
+	}
+	pbs_strncpy(fullhost, line, sizeof(fullhost));
+	strcat(fullhost, ".");
```

Comment:
If the only use of domain is here, why not keep the dot?  You initialize the for loop doing dot++.  If you only use it here, just copy in the dot.  That way you won't have to strcat() it here.

Reply:
I dont know,  some guess will be it calls gethostname on domainname,  leading dot might cause some trouble. I dont have the whole picture of the usecase, as it works, i decide to leave it unchanged.